src/main/java/edu/tamu/app/service/AbstractCatalogService.java

```diff
@@ -97,15 +103,47 @@ public void setProtocol(String protocol) {
     }
 
     protected String getAPIBase() {
-        return getProtocol() + "://" + getHost() + ":" + getPort() + "/" + getApp() + "/";
+        StringBuilder builder = new StringBuilder();
+
+        if (StringUtils.isNotEmpty(getProtocol())) {
```

Comment:
Should we default to http?

Reply:
I decided that since I was making it optional that when not specified it would default to "client default" behavior which is "//www.example.com/" and that would result in the appropriate http/https usage as decided by the client. That said, this behavior may not be necessary and we could just default to http.
(I also do not know if this behavior is actually in the standard or not.)